src/components/Modal/Modal.tsx

```diff
@@ -0,0 +1,42 @@
+import { useRef, useState } from "react";
+import ModalBasic from "./ModalBasic/ModalBasic";
+import styles from "./Modal.module.scss";
+import { useOnClickOutside } from "usehooks-ts";
+
+interface Modal {
+  name: string;
+  subtitle?: string;
+  shareButtons?: boolean;
+}
+
+function Modal({ name, ...rest }: Modal) {
+  // 평상시 modal 노출 false
+  const [isOpen, setIsOpen] = useState(false);
+  const modalRef = useRef<HTMLDivElement>(null);
+  // 모달 닫기 false
+  const handleClickOutside = () => {
+    setIsOpen((preIsOpen) => !preIsOpen);
+  };
```

Comment:
prev까지 표현해주는게 조금 더 명확할 것 같아요~
```suggestion
  const handleClickOutside = () => {
    setIsOpen((prevIsOpen) => !prevIsOpen);
  };
```